mathesar_ui/src/App.svelte

```diff
@@ -19,21 +19,28 @@
   import { modal } from './stores/modal';
   import { setReleasesStoreInContext } from './stores/releases';
   import ModalRecordSelector from './systems/record-selector/ModalRecordSelector.svelte';
-  import { loadLocaleAsync } from './i18n/i18n-load';
   import { setLocale } from './i18n/i18n-svelte';
+  import type { Locales } from './i18n/i18n-types';
+  import { loadTranslationsIntoMemory } from './i18n/i18n-load';
 
-  /**
-   * Later the translations file will be loaded
-   * in parallel to the FE's first chunk
-   */
   let isTranslationsLoaded = false;
-  (() => {
-    void loadLocaleAsync('en').then(() => {
-      setLocale('en');
+  const checkTranslationsInterval = setInterval(() => {
+    // eslint-disable-next-line prefer-destructuring
+    const translations:
+      | { lang: Locales; translationStrings: string }
+      | undefined =
+      // @ts-expect-error added by index.html
+      window.translations;
+    if (translations) {
+      loadTranslationsIntoMemory(
+        translations.lang,
+        JSON.parse(translations.translationStrings),
+      );
+      setLocale(translations.lang);
       isTranslationsLoaded = true;
-      return true;
-    });
-  })();
+      clearInterval(checkTranslationsInterval);
+    }
+  }, 100);
```

Comment:
This approach seems strange to me. If I understand correctly, in `index.html` we're importing the translation data and sticking it on `window`. Then here we are looking at `window` every 100ms to see if it has loaded. This seems really hacky. Why can't we do `await import` and let vite sort it out?

I wouldn't expect us to need to deal with `window`. But if we absolutely _do_, then I'd prefer to use a `.d.ts` file to provide our own typings for `window.translation` instead of the `@ts-expect-error` approach here.

Reply:
If we do `await import` vite will download `App.svelte` first & then start the download for the translations. The code here enables the parallel download for both of the things. 

I have added the `translations` object in `global.d.ts` #855da7b

Reply:
With the latest set of changes, this approach is no longer used.